.azure-pipelines/ultimate-pipeline.yml

```diff
@@ -842,21 +872,35 @@ stages:
             command: "CheckBuildLogsForErrors"
 
 - stage: exploration_tests_windows
-  condition: and(succeeded(), eq(variables['isScheduledBuild'], 'False'), eq(variables['runExplorationTests'], 'True'))
-  dependsOn: build_windows
+  condition: and(succeeded(), eq(variables['isScheduledBuild'], 'False'), or(eq(stageDependencies.check_changes.outputs['check_changes_job.check_changes_step.IsProfilerChanged'],'True'), eq(stageDependencies.check_changes.outputs['check_changes_job.check_changes_step.IsDebuggerChanged'], 'True')))
+  dependsOn: [build_windows, check_changes]
 
   jobs:
   - template: steps/update-github-status-jobs.yml
     parameters:
       jobs: [ExplorationTest]
 
   - job: generator
+    variables:
+      ISDEBUGGERCHANGED: $[stageDependencies.check_changes.check_changes_job.outputs['check_changes_step.IsDebuggerChanged']]
+      ISPROFILERCHANGED: $[stageDependencies.check_changes.check_changes_job.outputs['check_changes_step.IsProfilerChanged']]
     pool:
       vmImage: ubuntu-18.04
+
     steps:
       - bash: |
           matrix="{"
-          for explorationTestUseCase in "debugger" "continuousProfiler"; do
+
+          declare -a explorationTestUseCases=()
+          if [ $ISDEBUGGERCHANGED == "True" ]; then 
+            explorationTestUseCases+=("debugger")
+          fi
+
+          if [ $ISPROFILERCHANGED == "True" ]; then 
+            explorationTestUseCases+=("continuousProfiler")
+          fi
+
+          for explorationTestUseCase in ${explorationTestUseCases[@]}; do
```

Comment:
You're making me want to turn these generators into nuke targets 😛

Reply:
Moved to nuke